### PR TITLE
HDDS-9955. Simplify assertions in integration tests

### PR DESCRIPTION
**Pull request by wzhallright**
## What changes were proposed in this pull request?
Assertions in the form:  
``` 
assertTrue(<x> == <y>)   
assertEquals(null, <x>) or assertTrue(<x> == null)
assertTrue(<x> != null)
assertEquals(<boolean>, <x>)
```
can be simplified to:   
```
assertEquals(<x>, <y>)
assertNotNull(<x>)
assertTrue/assertFalse(<x>)
assertNull(<x>)
```
Not only are these simpler, some of them also provide more information in case of failure.

This PR updates such assertions in hadoop-ozone/integration-test.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9955
## CI
https://github.com/wzhallright/ozone/actions/runs/7284783456

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestOzoneRpcClientAbstract.java [hunk 1]
```diff
@@ -3796,7 +3796,7 @@ private void doMultipartUpload(OzoneBucket bucket, String keyName, byte val,
     assertTrue(latestVersionLocations.isMultipartKey());
     latestVersionLocations.getBlocksLatestVersionOnly()
         .forEach(omKeyLocationInfo ->
-            assertTrue(omKeyLocationInfo.getPartNumber() != -1));
+            assertNotEquals(omKeyLocationInfo.getPartNumber(), -1));
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestOzoneRpcClientAbstract.java:3799]:
```suggestion
            assertNotEquals(-1, omKeyLocationInfo.getPartNumber()));
```

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmBlockVersioning.java [hunk 2]
```diff
@@ -144,7 +143,7 @@ public void testAllocateCommit() throws Exception {
     List<OmKeyLocationInfo> locationInfoList =
         openKey.getKeyInfo().getLatestVersionLocations()
             .getBlocksLatestVersionOnly();
-    assertTrue(locationInfoList.size() == 1);
+    assertEquals(locationInfoList.size(), 1);
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmBlockVersioning.java:146]:
```suggestion
    assertEquals(1, locationInfoList.size());
```

**Review by adoroszlai — COMMENTED** [review]
Thanks @wzhallright for the patch.

**Comment by wzhallright** [comment]
@adoroszlai Thank you for your detailed comments and suggestions, has fixed the comments, PTAL.

**Review by adoroszlai — APPROVED** [review]

**Pull request merged**